optimum/intel/neural_compressor/neural_coder_adaptor.py

```diff
@@ -0,0 +1,26 @@
+class NeuralCoderAdaptor:
+    """API design adaption for Neural Coder"""
+
+    default_quant_dynamic = f"""\
+def eval_func(model):
+    EVAL_FUNC_LINES
+from neural_compressor.config import PostTrainingQuantConfig
+from optimum.intel.neural_compressor import INCQuantizer
+quantization_config = PostTrainingQuantConfig(approach="dynamic")
+quantizer = INCQuantizer.from_pretrained(MODEL_NAME)
+quantizer.quantize(quantization_config=quantization_config, save_directory="quantized_model", save_onnx_model=False)
+MODEL_NAME = quantizer._quantized_model
+MODEL_NAME.eval()
+"""
+
+    default_quant_static = f"""\
+def eval_func(model):
```

Comment:
Why do we need an `eval_func` in our case as it doesn't seem to be used ?

Reply:
Thanks for raising this. We seem to have missed referencing ```eval_func``` in the API. @PenghuiCheng Do we use ```eval_func``` like this in new Optimum API? ```INCQuantizer.from_pretrained(MODEL_NAME, eval_fn=eval_func)```, or do we use it some other way? Thanks

Reply:
@kaiyaointel， there are two usages： 1. user only provides eval_dataloader and metric then it will quantize with built-in eval_func, 2. user can provide an eval_func to INCQuantizer to tune the quantized model.

Reply:
OK, thanks, I will modify accordingly

Reply:
@echarlaix , I found that there is no built-in eval_func in INCQuantizer, so all examples didn't been tuned by  neural_compressor,  Why INCQuantizer didn't provide a built-in eval_func ?

Reply:
Sure, I can include the `eval_fn` in the `neural_compressor` examples.